### PR TITLE
CON-305 to release branch

### PR DESCRIPTION
**Pull request by Alxe**
Pulling most recent changes to documentation to the release branch.

### CODE REVIEW

File: docs/release_notes.rst [hunk 1]
```diff
@@ -19,13 +19,11 @@ macOS® platforms. It has been tested on the following systems:
 **Linux**
   * CentOS™ 7.0 (x64)
   * Red Hat® Enterprise Linux 7, 7.3, 7.5, 7.6, 8 (x64)
```
Comment (elainerti) [inline docs/release_notes.rst:21]: Let's add Red Hat 9 (it is in our 7.3.0 PAM)

File: docs/release_notes.rst [hunk 2]
```diff
@@ -19,13 +19,11 @@ macOS® platforms. It has been tested on the following systems:
...
-  * SUSE® Linux Enterprise Server 12 SP2 (x64)
   * Ubuntu® 18.04 (x64, Arm v7, Arm v8)
   * Ubuntu 20.04, 22.04 LTS (x64)
```
Comment (elainerti) [inline docs/release_notes.rst:23]: We also support Ubuntu 22.04 LTS on Arm v8

Comment (elainerti) [reply docs/release_notes.rst:23]: In theory, all the Ubuntu version should say LTS
Ubuntu 18.04 LTS
Ubuntu 20.04 LTS
Ubuntu 22.04 LTS

File: docs/release_notes.rst [hunk 3]
```diff
@@ -19,13 +19,11 @@ macOS® platforms. It has been tested on the following systems:
...
 
 **macOS**
-  * macOS 10.13-10.15, 12 (x64)
-  * macOS 11 (x64 and Arm v8 tested via x64 libraries)
+  * macOS 11, 12, 13 (x64)
 
 **Windows**
   * Windows 10, 11 (x64)
```
Comment (elainerti) [inline docs/release_notes.rst:29]: Do we need to list Windows Server 2016 and 2022 here?

**Comment by elainerti** [comment]
License.pdf is NOT matching the one we used in core. We are using Software License Agreement - #4000 (Rev. 01/2024)

**Comment by elainerti** [comment]
I think README is wrong. I don't think we should mention Python in the following paragraph:

License
RTI Connector for JavaScript and RTI Connector for Python are part of the Connext DDS Professional Package. If you have a valid license for the RTI Connext DDS Professional Package, such license shall govern your use of RTI Connector for Python and RTI Connector for JavaScript. All other use of this software shall be governed solely by the terms of RTI’s Software License for Non-Commercial Use #4040, included at the top level of the `Connector for Python repository

File: docs/release_notes.rst (hunk 1 above)
Comment (alexcamposruiz) [inline docs/release_notes.rst:21]:
```suggestion
  * Red Hat® Enterprise Linux 7, 7.3, 7.5, 7.6, 8, 9 (x64)
```

File: docs/release_notes.rst (hunk 2 above)
Comment (alexcamposruiz) [inline docs/release_notes.rst:23]:
```suggestion
  * Ubuntu 20.04 (x64), 22.04 (x64, Arm v8)
```

**Review by alexcamposruiz — APPROVED** [review]

**Pull request merged**